src/raft/raft.c

```diff
@@ -303,3 +305,30 @@ static int ioFsmVersionCheck(struct raft *r,
 
 	return 0;
 }
+
+int raft_io_describe_last_entry(struct raft_io *io,
+				raft_index *index,
+				raft_term *term)
+{
+	raft_term current_term;
+	raft_id voted_for;
+	struct raft_snapshot *snapshot;
+	raft_index start_index;
+	struct raft_entry *entries;
+	size_t n_entries;
+	int rv;
+
+	rv = io->load(io, &current_term, &voted_for, &snapshot,
+		      &start_index, &entries, &n_entries);
+	if (rv != 0) {
+		return rv;
+	}
+	POST(ERGO(n_entries == 0, snapshot != NULL));
```

Comment:
This is going to crash when `raft_io_describe_last_entry` is called and there are no entries in the raft log + snapshots, correct? If that is the case, should this be a crash or an empty return?

Reply:
Good point, thanks, this could potentially be used with the data directory of a node that never got the chance to write a log entry to disk (when bootstrapping we always write the entry with the initial configuration, but joiner nodes can start with nothing at all). In this case we should probably return `(0, 0)` since that will compare less than the tuple for any node that does have entries on disk.

Reply:
Fixed in the latest commit.